core/src/main/scala/zio/config/ConfigDescriptorModule.scala

```diff
@@ -202,40 +202,93 @@ trait ConfigDescriptorModule extends ConfigSourceModule { module =>
           )
       )
 
-    def set[K, V, A](desc: ConfigDescriptor[A]): ConfigDescriptor[Set[A]] =
+    def set[K, V, A](desc: => ConfigDescriptor[A]): ConfigDescriptor[Set[A]] =
       list(desc).xmapEither(distinctListToSet, s => Right(s.toList))
 
-    def set[A](path: K)(desc: ConfigDescriptor[A]): ConfigDescriptor[Set[A]] =
+    def set[A](path: K)(desc: => ConfigDescriptor[A]): ConfigDescriptor[Set[A]] =
       nested(path)(set(desc))
 
     private def distinctListToSet[A](list: List[A]): Either[String, Set[A]] =
       if (list.size == list.distinct.size) Right(list.toSet) else Left("Duplicated values found")
 
   }
 
+  case class LazyConfigDescriptor[A](get: () => ConfigDescriptor[A]) {
+    def map[B](f: ConfigDescriptor[A] => ConfigDescriptor[B]): LazyConfigDescriptor[B] =
+      LazyConfigDescriptor(() => f(get()))
+  }
+
+  final def thunk[A](config: => ConfigDescriptor[A]): LazyConfigDescriptor[A] =
+    LazyConfigDescriptor(() => config)
+
+  def dump[A](config: ConfigDescriptor[A]): String = {
+    val builder = new StringBuilder
+
+    def go[A](config: ConfigDescriptor[A], prefix: String): Unit =
+      config match {
+        case Default(config, default) =>
+          builder.append(s"${prefix}Default($default}\n")
+          go(config.get(), prefix + "  ")
+        case Describe(config, message) =>
+          builder.append(s"${prefix}Describe($message}\n")
+          go(config.get(), prefix + "  ")
+        case DynamicMap(_, config) =>
+          builder.append(s"${prefix}DynamicMap\n")
+          go(config.get(), prefix + "  ")
+        case Nested(_, path, config) =>
+          builder.append(s"${prefix}Nested($path)\n")
+          go(config.get(), prefix + "  ")
+        case Optional(config) =>
+          builder.append(s"${prefix}Optional\n")
+          go(config.get(), prefix + "  ")
+        case OrElse(left, right) =>
+          builder.append(s"${prefix}OrElse\n")
+          go(left.get(), prefix + "  ")
+          go(right.get(), prefix + "  ")
+        case OrElseEither(left, right) =>
+          builder.append(s"${prefix}OrElseEither\n")
+          go(left.get(), prefix + "  ")
+          go(right.get(), prefix + "  ")
+        case Sequence(_, config) =>
+          builder.append(s"${prefix}Sequence\n")
+          go(config.get(), prefix + "  ")
+        case Source(_, propertyType) =>
+          builder.append(s"${prefix}Source($propertyType)\n")
+          ()
+        case Zip(left, right) =>
+          builder.append(s"${prefix}Zip\n")
+          go(left.get(), prefix + "  ")
+          go(right.get(), prefix + "  ")
+        case XmapEither(config, _, _) =>
+          go(config.get(), prefix)
+      }
+    go(config, "")
+    builder.toString()
```

Comment:
I did not want to leave it in the code first, but then had to re-add several times while debugging and thought that it may help people in future development. Now that the descriptors are lazy, without this it was very inconvenient to examine a descriptor subtree in debugger / in debug logs.
Should I remove it?